### ISSUE TITLE
printf on embedded platform

### ISSUE THREAD

**Issue by ajraymond** [bluekitchen/btstack#156]
Two issues, because my platform doesn't implement printf:
1) btstack_debug.h defines BTSTACK_PRINTF, with the following comment `// allow to provide port specific printf`, which is great, but it is used nowhere in btstack.

2) some modules, including hsp_hs.c, use printf directly instead of, for example, log_info().

**Comment by mringwal** [comment]
good point about the printf thing. I've recently stranded on a platform, where the vendor provided makefiles uses glibc instead of nano or so and was puzzle why printf wasn't working...

BTSTACK_PRINTF is quite ugly. I guess a bstack_printf(..) would be better. And yes, it should get used in the examples.

printf is not supposed to show up anywhere in src/ - I'll do another sweep.
